src/dispdrivers/vgacontroller.cpp

```diff
@@ -151,33 +151,95 @@ void IRAM_ATTR VGAController::VSyncInterrupt(void * arg)
 }
 
 
+std::function<uint8_t(RGB888 const &)> VGAController::getPixelLamda(PaintMode mode)
+{
+  switch (mode) {
+    case PaintMode::XOR:
+      return [&] (RGB888 const & color) { return preparePixel(color) & 63; };
+    case PaintMode::ANDNOT:
+    case PaintMode::ORNOT:
+      return [&] (RGB888 const & color) { return (~preparePixel(color) & 63) || m_HVSync; };
+    default: // PaintMode::Set, et al
+      return [&] (RGB888 const & color) { return preparePixel(color); };
+  }
+}
+
+std::function<void(int X, int Y, uint8_t pattern)> VGAController::setPixelLamda(PaintMode mode)
+{
+  switch (mode) {
+    case PaintMode::Set:
+      return [&] (int X, int Y, uint8_t pattern) { VGA_PIXEL(X, Y) = pattern; };
+    case PaintMode::OR:
+    case PaintMode::ORNOT:
+      return [&] (int X, int Y, uint8_t pattern) { VGA_PIXEL(X, Y) |= pattern; };
+    case PaintMode::AND:
+    case PaintMode::ANDNOT:
```

Comment:
NOT variants for setting pixels and filling them use their regular counterparts, as the pattern will be pre-inverted